Refactor project structure, add image preview, center banner

ojasvib opened the pull request:
- Added preview functionality for added images
- Added "Go" button to initiate image generation process
- Centered the generated banner for better visual alignment
- Moved files for improved project structure and organization

![image](https://github.com/user-attachments/assets/5e2c764f-56a0-4175-a8a8-d0720422dcfa)

ojasvib committed:
Refactor project structure, add image preview, center banner, and improve layout

- Added preview functionality for newly added images
- Implemented "Go" button to initiate image generation process
- Centered the generated banner for better visual alignment
- Moved files for improved project structure and organization

ojasvib committed:
Hide keyboard on 'Go' button click in InputField composable

- Added functionality to hide the keyboard when the 'Go' button is clicked

sanskar10100 commented on app/src/main/java/com/ojasvi/bannr/ui/screens/PhotoReasoningScreen.kt:
```diff
@@ -31,46 +34,50 @@ import androidx.compose.runtime.saveable.rememberSaveable
     modifier: Modifier = Modifier.navigationBarsPadding(),
     viewModel: PhotoReasoningViewModel = viewModel()
 ) {
+    var productImagesUri by rememberSaveable { mutableStateOf<List<Uri>>(listOf()) }
```
Consider a SnapshotStateList to avoid making copies of the list when updating the state. Also consider if this should be in the ViewModel (I believe it should be)

ojasvib replied:
You're correct. Shifted it to the ViewModel and used SnapshotStateList (which is mutableStateListOf in the ViewModel).

sanskar10100 commented on app/src/main/java/com/ojasvi/bannr/ui/screens/PhotoReasoningScreen.kt:
```diff
@@ -164,59 +178,62 @@ private fun InputField(
+                        contentDescription = null,
+                        modifier = Modifier
+                            .padding(4.dp)
+                            .requiredSize(72.dp)
```
Why `requiredSize` instead of the regular `size`?

ojasvib replied:
I erred.

sanskar10100 reviewed (approved):
Generally good, consider feedback before merging.

ojasvib committed:
Refactor productImagesUri to ViewModel as SnapshotStateList and switch to .size for image items

- Moved productImagesUri to ViewModel as a SnapshotStateList for efficient state management
- Replaced .requiredSize with .size for image items in LazyRow to allow flexible sizing based on layout constraints
